### DISCUSSION TITLE
Allow filter for Branches/Tags in Manual Workflow selection

### DISCUSSION THREAD

**Discussion by derN3rd** [community/community#32730] [Actions]
Hey there,

I'm currently trying to setup our deployment process via Actions, but I could not find any setting regarding the selection of which branch/tag I want to allow for this workflow_dispatch action.

From the [trigger docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch) I know that there is currently only the inputs option for workflow_dispatch and I know I can restrict and fail a job if it's run under a wrong condition (i.e. by adding a check as first step), but I would be great if we can setup something like

```
on:
  workflow_dispatch:
    tags: v*
    branches:
```

to only allow this manual trigger to be run on tags or filter by specific naming.

Thanks in advance

**Comment by kylemclaren** [comment]
👍  this is a requirement of mine too.

I would like to specify which branches a `workflow` dispatch workflow can be run on. ie. `main` or `branch/*` etc. It should not be possible to run the workflow on non-matching branches.

**Comment by sishbi** [comment]
👍🏻 This is also my desire.  
Another request would be to add the input parameter values to the workflow name in the actions list when it is run, but that's another issue.
